nixos/modules/profiles/hardened.nix

```diff
@@ -14,12 +14,17 @@ with lib;
 
   nix.allowedUsers = mkDefault [ "@users" ];
 
+  environment.memoryAllocator.provider = mkDefault "scudo";
+  environment.variables.SCUDO_OPTIONS = mkDefault "ZeroContents=1";
+
   security.hideProcessInformation = mkDefault true;
 
   security.lockKernelModules = mkDefault true;
 
   security.allowUserNamespaces = mkDefault false;
 
+  nix.useSandbox = mkDefault false;
```

Comment:
Are usernamespaces also not allowed for privileged users?

Reply:
Debian (used to?) carry a patch that allowed disabling unprivileged ns only.  Vanilla kernel is all or nothing, iirc.

Reply:
Should security.allowUserNamespaces imply nix.useSandbox?

Reply:
Here is the relevant debian [patch](https://salsa.debian.org/kernel-team/linux/-/blob/master/debian/patches/debian/add-sysctl-to-disallow-unprivileged-CLONE_NEWUSER-by-default.patch). If allowUserNamespaces is false it should force useSandbox to false as well, if this is a change we want. Ideally I would prefer to rewrite a more scoped patch for the kernel that was a restricted user namespace patch to only allow the build user group and maybe one additional group so namespaces could be used additionally by say a user wanting to benefit from giving chromium/other applications access to the user namespaces but not giving it system wide to every user.